### DISCUSSION TITLE
Applying a specific hierarchy preset to a dropped file

### DISCUSSION THREAD

**Discussion by Marwa-Da** [molstar/molstar#1033] [Q&A]
Hi all,

When a file is dropped to the viewer or loaded using the home left panel, the plugin automatically applies the default representation preset set earlier in the configuration. However, if there's a need to apply a specific **hierarchy preset** instead of the default one, how can this be achieved?

**Comment by arose** [comment]
You can change the default preset, there is a config option see https://github.com/molstar/molstar/blob/4b4f6d34a341d867fd23f5352c37df0f52f48371/src/apps/viewer/app.ts#L193

**Reply by Marwa-Da** [reply]
@arose I think you misunderstood me, I want to apply a specific hierarchy preset. The `DefaultRepresentationPreset `is working perfectly!

**Comment by Marwa-Da** [comment]
@dsehnal Any thoughts about it would be appreciated.

**Reply by dsehnal** [reply]
I feel like Alex already answered this. Just change the default preset to whatever you want and it will be applied instead.

**Reply by Marwa-Da** [reply]
I'm not using DefaultRepresentationPreset, I created another one and it's getting applied on dropping a file perfectly. But I'm complaining about showing all models at once instead of showing one model with the navigator.
![image](https://github.com/molstar/molstar/assets/11466297/342f43d3-6075-4c42-9845-ef5f4f5675b8)

**Reply by Marwa-Da** [reply]
Showing all models at once when a file is dropped to the viewer or loaded using the home left panel.

**Reply by dsehnal** [reply]
There is an "all models" preset (`preset-trajectory-all-models`) that does that.

----------

Also, you are getting quite a lot of help in developing your app @Marwa-Da . What do you think about contributing something back to the project? For example improving documentation?

**Reply by Marwa-Da** [reply]
Sure!! I forked the molstar-cocs repo and I'll create a PR
